CRM/Core/BAO/CustomField.php

```diff
@@ -1223,10 +1223,18 @@ private static function formatDisplayValue($value, $field, $contactID = NULL) {
             $display = $url['file_url'];
           }
         }
-        // In other contexts show a paperclip icon
         elseif ($value) {
-          $icons = CRM_Core_BAO_File::paperIconAttachment('*', $value);
-          $display = $icons[$value];
+          // In the context of inline view, show file/image
+          if (is_string($value)) {
```

Comment:
Nope :(  In case of inline view for customField of File type, uri got passed instead of expected file ID and it starts from here https://github.com/civicrm/civicrm-core/blame/master/CRM/Core/BAO/CustomGroup.php#L1894 and eventually causes type-check-validation-error as mentioned in the description. 

Also in case of inline view I think its not appropriate to call paperIconAttachment() rather than showing the filename as link as it was in prior version, reason why have to call  getEntityFile() instead.